File: test-suite-caffeine-kotlin/src/test/kotlin/io/micronaut/cache/docs/MyRemovalHandler.kt
package io.micronaut.cache.docs

import com.github.benmanes.caffeine.cache.RemovalCause
import jakarta.inject.Singleton

@Singleton
class MyRemovalHandler {

    val removals: MutableList<String> = ArrayList()

    fun handle(key: String, value: Int, cause: RemovalCause) {
        removals.add("$key|$value|$cause")
    }
}

File: test-suite-caffeine-python/src/test/python/micronaut/cache/docs/MyRemovalHandler.py
from com.github.benmanes.caffeine.cache import RemovalCause
from jakarta.inject import Singleton


@Singleton
class MyRemovalHandler:

    def __init__(self):
        self.removals: list[str] = []

    def handle(self, key: str, value: int, cause: RemovalCause) -> None:
        self.removals.append(f"{key}|{value}|{cause}")

    def get_removals(self) -> list[str]:
        return self.removals
